libs/core/src/lib/checkbox/checkbox/checkbox.component.spec.ts

```diff
@@ -0,0 +1,198 @@
+import {async, ComponentFixture, TestBed, tick} from '@angular/core/testing';
+
+import {CheckboxComponent} from './checkbox.component';
+import {FormsModule} from '@angular/forms';
+import {Component, ViewChild} from '@angular/core';
+
+@Component({
+    template: `
+        <fd-checkbox [(ngModel)]="value"></fd-checkbox>`
+})
+class TestCheckboxComponent {
+    @ViewChild(CheckboxComponent, {static: false}) checkboxRef;
+    value: any = false;
+}
+
+describe('CheckboxComponent', () => {
+    let checkbox: CheckboxComponent;
+    let hostComponent: TestCheckboxComponent;
+    let fixture: ComponentFixture<TestCheckboxComponent>;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [FormsModule],
+            declarations: [CheckboxComponent, TestCheckboxComponent]
+        });
+
+        fixture = TestBed.createComponent(TestCheckboxComponent);
+        hostComponent = fixture.componentInstance;
+        fixture.detectChanges();
+        checkbox = fixture.componentInstance.checkboxRef;
+    });
+
+    it('should create', () => {
+        expect(hostComponent).toBeTruthy();
+        expect(checkbox).toBeTruthy();
+    });
+
+    it('should properly bind control value', () => {
+        fixture.detectChanges();
+
+        spyOn(checkbox, 'writeValue');
+
+        hostComponent.value = true;
+        fixture.detectChanges();
+
+        hostComponent.value = false;
+        fixture.detectChanges();
+
+        expect(checkbox.writeValue).toHaveBeenCalledTimes(3);
+        expect(checkbox.checkboxValue).toBe(false);
+    });
+
+    // it('should be unchecked on double click', () => {
```

Comment:
What are all of these comments for. IF they are necessary remove the ones with nothing inside

Reply:
Test cases have been updated